src/collection-browser.ts

```diff
@@ -1702,10 +1724,12 @@ export class CollectionBrowser
     // When we reach the end of the data, we can set the infinite scroller's
     // item count to the real total number of results (rather than the
     // temporary estimates based on pages rendered so far).
-    if (results.length < this.pageSize) {
+    const resultCountDiscrepancy = numRows - results.length;
+    if (resultCountDiscrepancy > 0) {
       this.endOfDataReached = true;
       if (this.infiniteScroller) {
-        this.infiniteScroller.itemCount = this.totalResults;
+        this.infiniteScroller.itemCount =
+          this.estimatedTileCount - resultCountDiscrepancy;
```

Comment:
This is to work around some `filter-top-level` issues on collections. Because the `totalResults` value sent by the PPS doesn't always match the actual number of results it sends, we can't rely on it to cap off the infinite scroller's item count.